doc/ARS/Local_ARS_HLD.md

```diff
@@ -0,0 +1,1052 @@
+    * If "set" capability is not supported - ignore NHG update.
+    * If an alternative path is defined:
+        - Identify the relevant nexthops by matching them against the entries in the ARS_NEXTHOP_GROUP_MEMBER table with the role set to "alternative_path."
+        - Set these nexthops as part of the alternative path by updating the SAI attributes for the nexthop group or member accordingly.
+5. Add ROUTE_TABLE Table
+    * The APPL_DB adds the ROUTE_TABLE table, triggering the process.
+6. Route creation
+    * RouteOrch creates/updates route with NHG reference.
+7. Notification from RouteOrch
+    * RouteOrch component will notify ArsOrch of NHG update. ArsOrch will check if the NHG is implicit, meaning it was created by NhgOrch and will ignore notification
+
+##### __Figure 8: ARS LAG table creation flow__
+![](images/ars_lag_create.png)
+
+1. Add ARS_PORTCHANNEL Table
+    * The CONFIG_DB adds the ARS_PORTCHANNEL table, triggering the process.
+2. Create SAI ARS Object
+    * The ArsOrch component receives the update and creates the SAI ARS object.
+4. LAG's SAI attributes handling:
+    * If LAG was created:
+        - It is updated with the ARS object ID.
+
+##### __Figure 9: LAG creation flow__
+![](images/lag_create.png)
+
+1. LAG creation
+    * PortsOrch receices LAG from APPL_DB, triggering the process.
+2. Notification
+    * PortsOrch component will notify ArsOrch of LAG creation.
+4. LAG's SAI attributes handling:
+    * LAG is updated with the ARS object ID.
+
+##### __Figure 10: LAG member addition flow__
+![](images/lag_update.png)
+
+1. LAG change
+    * PortsOrch receices LAG member from APPL_DB, triggering the process.
+2. Notification
+    * PortsOrch component will notify ArsOrch of LAG member change.
+4. LAG's SAI attributes handling:
+    * If an alternative path is defined
+        - Set the relevant port as part of the alternative path.
+
+##### __Figure 11: ACL configuration flow__
+![](images/acl_config.png)
+
+1. Users define custom ACL table type in ACL_TABLE_TYPE with DISABLE_ARS_FORWARDING type.
+2. ACL table added, referencing the custom table type.
+3. ACL rule added, referencing ARS disable action.
+
+### SAI API 
+
+There are no new SAI APIs or attributes required.
+
+Following table lists SAI usage and supported attributes with division to phase implementation:
+
+* Phase 1
+
+| SAI api | Supported SAI attribute 
+| ------- | ----------------------- 
+|create_ars_profile | SAI_ARS_PROFILE_ATTR_PORT_LOAD_PAST<br>SAI_ARS_PROFILE_ATTR_LOAD_PAST_MIN_VAL<br>SAI_ARS_PROFILE_ATTR_LOAD_PAST_MAX_VAL<br>SAI_ARS_PROFILE_ATTR_ENABLE_IPV4<br>SAI_ARS_PROFILE_ATTR_ENABLE_IPV6<br>SAI_ARS_PROFILE_ATTR_MAX_FLOWS<br>SAI_ARS_PROFILE_ATTR_ALGO<br>SAI_ARS_PROFILE_ATTR_PORT_LOAD_FUTURE<br>SAI_ARS_PROFILE_ATTR_PORT_LOAD_PAST_WEIGHT<br>SAI_ARS_PROFILE_ATTR_PORT_LOAD_FUTURE_WEIGHT<br>SAI_ARS_PROFILE_ATTR_SAMPLING_INTERVAL<br>SAI_ARS_PROFILE_ATTR_ECMP_ARS_MAX_GROUPS<br>SAI_ARS_PROFILE_ATTR_ECMP_ARS_MAX_MEMBERS_PER_GROUP

```

Comment:
How's the ARS profile binding to a switch handled? refer the section 6.2 in ARS SAI spec.
https://github.com/opencomputeproject/SAI/blob/master/doc/ARS/Adaptive-Routing-and-Switching.md#62-bind-ars-profile-object-to-switch-object